src/miner.cpp

```diff
@@ -298,6 +311,43 @@ void BlockAssembler::SortForBlock(const CTxMemPool::setEntries& package, std::ve
     std::sort(sortedEntries.begin(), sortedEntries.end(), CompareTxIterByAncestorCount());
 }
 
+CFeeRate BlockAssembler::minTxFeeRate()
+{
+    int packages_selected = 0;
+    int descendants_updated = 0;
+    CFeeRate min_fee_rate = CFeeRate(COIN, 1);
+
+    resetBlock();
+    pblocktemplate.reset(new CBlockTemplate());
+
+    if (!pblocktemplate.get()) {
+        return min_fee_rate;
+    }
+    CBlock* block = &pblocktemplate->block;
+
+    // Add dummy coinbase tx as first transaction
+    block->vtx.emplace_back();
+    pblocktemplate->vTxFees.push_back(-1);
+    pblocktemplate->vTxSigOpsCost.push_back(-1);
```

Comment:
Are these definitely needed? You're just going to throw the block template away at the end.

Reply:
I think you're right, so I've removed these three lines. nice catch!